### ISSUE TITLE
gps ls errors out when a branch has multiple commits on it

### ISSUE THREAD

**Issue by drewdeponte** [uptech/git-ps-rs#71]
Currently if a remote branch that was linked to a patch originally has multiple commits on it. gps ls will bail out with an error as part of processing.

This shouldn't happen. Instead it should continue but communicate the state that there are multiple commits on that remote branch some how and that things are out of sync and that they need to be investigated and addressed.

**Comment by drewdeponte** [comment]
I talked about this with @RyanHedges a bit tonight after thinking through it a bit.

The conclusion that we came to is that we should be tracking the diff hash when an operation occurs and state changes and that diff hash should be stored with that state.

This will then enable us to not only identify that things are different between local and remote but will allow us to tell if local has changed since we performed the operation or if the remote has changed since we performed the operation.

If the local has changed and the remote hasn't then we simply need to `sync` or `rr` our patch to update the remote.

If the local hasn't changed and the remote has then we need to investigate and potentially cherry-pick changes from the remote branch into our local patch stack and integrate them into our patch and then `sync` or `rr`.

If both the local and the remote have changed since the operation we need to potentially cherry-pick changes from the remote branch into our local patch stack and integrate them into our patch and then `sync` or `rr`.

Off the bat we were thinking we could use `+` to indicate local changes (e.g. `rr+`) and maybe use `!` or `!!` to indicate remote changes. Then we could have `rr+!` indicating that both changed.

How we visually represent these states is probably the least solidified portion of this. But given it is the easiest to change I am not too concerned about it.
